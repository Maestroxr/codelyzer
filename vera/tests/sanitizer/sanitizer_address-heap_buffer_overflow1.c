#include<stdio.h> 
#include<stdlib.h> 
int main(void) 
{ 
	int result = 0;
	char* p = (char*)malloc(3); 
	if (!p) 
	{
		result = -1;
	}
	int i = 0; 
	for(i=0;i<0Xffffffff;i++) 
	{ 
		p[ i ] = 'a'; 
	} 
	printf("\n %s \n", p); 
	return result; 
}