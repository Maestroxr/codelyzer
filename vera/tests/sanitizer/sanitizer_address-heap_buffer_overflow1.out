C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-heap_buffer_overflow1.c:14: Memory:[C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-heap_buffer_overflow1] Heap buffer overflow, suspect variables: ['i', 'p'], sanitizer_address-heap_buffer_overflow1.c:14
	Was accessed by write of size 1 at:
		sanitizer_address-heap_buffer_overflow1.c:14 in function 'main'
	Accessed address is located 0 bytes to the right of 3-byte region, allocated at:
		sanitizer_address-heap_buffer_overflow1.c:6 in function 'main'
:MemoryEnd:
